tensorflow/python/util/tf_inspect.py

```diff
@@ -45,6 +45,27 @@ def getargspec(object):  # pylint: disable=redefined-builtin
                if d.decorator_argspec is not None), _inspect.getargspec(target))
 
 
+def getfullargspec(object):  # pylint: disable=redefined-builtin
+  """TFDecorator-aware replacement for inspect.getargspec.
+
+  Args:
+    object: A callable, possibly decorated.
+
+  Returns:
+    The `FullArgSpec` (Python 3) or `ArgSpec` (Python 2)
```

Comment:
Are these classes compatible enough that all downstream code can use them interchangeably? I.e. is FullArgSpec backwards compatible with ArgSpec?